src/core/qgsweakrelation.h

```diff
@@ -134,6 +138,15 @@ class CORE_EXPORT QgsWeakRelation
      */
     QString referencingLayerName() const;
 
+    /**
+     * Sets the source for the referencing (or "child" / "right") layer, by \a source URI and \a provider ID.
+     *
+     * \see referencingLayerSource()
+     * \see referencingLayerProvider()
+     * \since QGIS 3.36
+     */
+    void setReferencingLayer( const QString &source, const QString &provider );
```

Comment:
Perhaps sourceURI to clarify the semantics? (if so the above Doxygen comment should be amended , and same remark for below methods)
```suggestion
    void setReferencingLayer( const QString &sourceURI, const QString &provider );
```